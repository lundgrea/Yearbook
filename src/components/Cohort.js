import React, { Component } from 'react';
import Person from './Person';
import people from '../data/yearbook-data.js';
import './Cohort.css';

class Cohort extends Component {
  constructor() {
    super();
    this.state = {}
  }

  mappedPeople = (type) => {
    return this.props[type].map(person => {
      return (<Person person={person}/>)
    })
  }

  render() {
  return (
    <section className="cohort-container">
      {this.mappedPeople('staff')}
      {this.mappedPeople('students')}
    </section>
  )
  }

}

export default Cohort;
