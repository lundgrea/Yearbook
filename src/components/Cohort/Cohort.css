.Cohort-container {
    padding: 20px;
    grid-template-columns: repeat(3, 1fr);
    grid-gap: 20px; 
  }