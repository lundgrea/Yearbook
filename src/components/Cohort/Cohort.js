import React from 'react';
import Person from '../Person/Person';
import PropTypes from 'prop-types'
import './Cohort.css';


const Cohort = ({title, data, deletePerson, updatePerson}) => {

  const mappedPeople = data.map(person => {
    return <Person key={person.id} id={person.id} photo={person.photo} name={person.name} quote={person.quote} superlative={person.superlative} deletePerson={deletePerson} updatePerson={updatePerson}/>
  })

  return (
    <section className="Cohort-container">
      <h2>{title}</h2>
      <div className="card">
        {mappedPeople}
      </div>
    </section>
  )

}

export default Cohort;

Cohort.propTypes = {
  staff: PropTypes.array,
  students: PropTypes.array
}
