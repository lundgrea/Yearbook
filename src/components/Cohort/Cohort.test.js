import React from 'react';
import { shallow } from 'enzyme';
import Cohort from './Cohort';

describe('Cohort', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const mockData = [{id: 1, name: 'Eric', quote: 'You miss all the shots you don\'t take.', superlative: 'Most Likely to Win the PGA', photo: './eric.jpg'},
    {id: 2, name: 'Amy', quote: 'I believe in you, cuties!', superlative: 'Most Likely to Be an Actual Superhero', photo: './amy.jpg'}
    ]
    const wrapper = shallow(<Cohort
        title="Staff"
        data={mockData}
        deletePerson={jest.fn()}
        updatePerson={jest.fn()}
      />);

    expect(wrapper).toMatchSnapshot();
  });
});