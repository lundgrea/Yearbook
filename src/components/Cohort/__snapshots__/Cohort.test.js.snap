// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cohort should match the snapshot with all data passed in correctly 1`] = `
<section
  className="Cohort-container"
>
  <h2>
    Staff
  </h2>
  <div
    className="cards"
  >
    <Person
      deletePerson={[MockFunction]}
      id={1}
      key="1"
      name="Eric"
      photo="./eric.jpg"
      quote="You miss all the shots you don't take."
      superlative="Most Likely to Win the PGA"
      updatePerson={[MockFunction]}
    />
    <Person
      deletePerson={[MockFunction]}
      id={2}
      key="2"
      name="Amy"
      photo="./amy.jpg"
      quote="I believe in you, cuties!"
      superlative="Most Likely to Be an Actual Superhero"
      updatePerson={[MockFunction]}
    />
  </div>
</section>
`;
